tfx_bsl/beam/run_inference.py

```diff
@@ -568,69 +644,84 @@ def _pre_process(self) -> _IOTensorSpec:
     # Pre process functions will validate for each signature.
     io_tensor_specs = []
     for signature in self._signatures:
-      if len(signature.signature_def.inputs) != 1:
-        raise ValueError('Signature should have 1 and only 1 inputs')
-      if (list(signature.signature_def.inputs.values())[0].dtype !=
+      if (len(signature.signature_def.inputs) == 1 and
+          list(signature.signature_def.inputs.values())[0].dtype !=
           tf.string.as_datatype_enum):
         raise ValueError(
-            'Input dtype is expected to be %s, got %s' %
+            'With 1 input, dtype is expected to be %s, got %s' %
             tf.string.as_datatype_enum,
             list(signature.signature_def.inputs.values())[0].dtype)
       io_tensor_specs.append(_signature_pre_process(signature.signature_def))
-    input_tensor_name = ''
-    input_tensor_alias = ''
+    input_tensor_names = []
+    input_tensor_alias = []
+    input_tensor_types = {}
     output_alias_tensor_names = {}
     for io_tensor_spec in io_tensor_specs:
-      if not input_tensor_name:
-        input_tensor_name = io_tensor_spec.input_tensor_name
+      if not input_tensor_names:
+        input_tensor_names = io_tensor_spec.input_tensor_names
         input_tensor_alias = io_tensor_spec.input_tensor_alias
-      elif input_tensor_name != io_tensor_spec.input_tensor_name:
+      elif input_tensor_names != io_tensor_spec.input_tensor_names:
         raise ValueError('Input tensor must be the same for all Signatures.')
-      for alias, tensor_name in io_tensor_spec.output_alias_tensor_names.items(
-      ):
+      for alias, tensor_type in io_tensor_spec.input_tensor_types.items():
+        input_tensor_types[alias] = tensor_type
+      for alias, tensor_name in io_tensor_spec.output_alias_tensor_names.items():
         output_alias_tensor_names[alias] = tensor_name
-    if (not output_alias_tensor_names or not input_tensor_name or
+    if (not output_alias_tensor_names or not input_tensor_names or
         not input_tensor_alias):
       raise ValueError('No valid fetch tensors or feed tensors.')
-    return _IOTensorSpec(input_tensor_alias, input_tensor_name,
-                         output_alias_tensor_names)
+    return _IOTensorSpec(input_tensor_alias, input_tensor_names,
+                         input_tensor_types, output_alias_tensor_names)
 
   def _has_tpu_tag(self) -> bool:
     return (len(self._tags) == 2 and tf.saved_model.SERVING in self._tags and
             tf.saved_model.TPU in self._tags)
 
+  def _extract_from_recordBatch(self, elements: pa.RecordBatch):
+    serialized_examples = bsl_util.ExtractSerializedExampleFromRecordBatch(elements)
+
+    model_input = None
+    if (len(self._io_tensor_spec.input_tensor_names) == 1):
+      model_input = {self._io_tensor_spec.input_tensor_names[0]: serialized_examples}
```

Comment:
Can we just leave this in _BaseBatchsavedModelDoFn and move the rest to _BatchPredictDoFn?